pyro/contrib/gp/models/model.py

```diff
@@ -136,20 +138,30 @@ def set_data(self, X, y=None):
 
         + Batch training on a sparse variational model:
 
+            .. doctest::
+               :hide:
+
+                >>> X = torch.tensor([[1., 5, 3], [4, 3, 7]])
+                >>> y = torch.tensor([2., 1])
+                >>> kernel = gp.kernels.RBF(input_dim=3)
+                >>> kernel.set_prior("variance", dist.Uniform(torch.tensor(0.5), torch.tensor(1.5)))
+                >>> kernel.set_prior("lengthscale", dist.Uniform(torch.tensor(1.0), torch.tensor(3.0)))
+                >>> optimizer = pyro.optim.Adam({"lr": 0.01})
+
             >>> Xu = torch.tensor([[1., 0, 2]])  # inducing input
             >>> likelihood = gp.likelihoods.Gaussian()
-            >>> vsgp = gp.models.SparseVariationalGP(X, y, kernel, Xu, likelihood)
+            >>> vsgp = gp.models.VariationalSparseGP(X, y, kernel, Xu, likelihood)
             >>> svi = SVI(vsgp.model, vsgp.guide, optimizer, Trace_ELBO())
             >>> batched_X, batched_y = X.split(split_size=10), y.split(split_size=10)
             >>> for Xi, yi in zip(batched_X, batched_y):
             ...     vsgp.set_data(Xi, yi)
-            ...     svi.step()
+            ...     svi.step()  # doctest: +SKIP
 
-        + Making a two-layer Gaussian Process stochastic function:
+        + Making a two-layer Gaussian Process stochastic function::
```

Comment:
Can you explain the `:` vs `::` rule you're following? This has always confused me, and I'd like to learn the right way.

Reply:
I think `:` will have no effect, and will just appear as is, so the question is whether to use `::` (which also appears as a regular `:`) or not.

 - If we have an interactive (>>>) block, we do not need `::`. If we put `::`, it will treat the `.. doctest` directive as part of the literal block, which we do not want. So with the interactive prompt, its always safer to use `:`.
 - If we don't have an interactive code block (no `>>>`), we need to use `::`, which will treat the code block as a block of python code, by default.

Reply:
Let me enforce some consistency in these code blocks actually.